src/lib/validators/unique-values-validator.ts

```diff
@@ -0,0 +1,31 @@
+import {
+    AbstractControl,
+    ValidationErrors,
+    ValidatorFn
+} from '@angular/forms';
+import { isNullOrUndefined } from 'util';
+
+export function uniqueValuesValidator(uniqueKeys?:Array<string>):ValidatorFn
+{
+    return (control:AbstractControl):ValidationErrors | null =>
+    {
+        let seen:Set<unknown> = new Set();
+        const hasDuplicates:boolean = (control.value as Array<unknown>).some((currentObject:unknown) =>
```

Comment:
würde hier eine andere Benennung vorschlagen
```suggestion
        const hasDuplicates:boolean = (control.value as Array<unknown>).some((value:unknown) =>
```